### PR TITLE
Switch from webidl to weedle in script_plugins

### PR DESCRIPTION
**Pull request by Eijebong**
This removes the dependency on lalrpop and should speed up compilation
quite a bit.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23405)

### CODE REVIEW

File: components/script_plugins/webidl_must_inherit.rs [hunk 1]
```diff
@@ -94,10 +92,22 @@ fn is_webidl_ty(cx: &LateContext, ty: &ty::TyS) -> bool {
...
+    let idl = weedle::parse(code).expect("Invalid webidl provided");
+    let mut inherits = "";
+
+    for def in idl {
+        if let weedle::Definition::Interface(def) = def {
+            if let Some(parent) = def.inheritance {
+                inherits = parent.identifier.0;
+                break;
+            }
+        } else if let weedle::Definition::CallbackInterface(def) = def {
```
Comment (CYBAI) [inline components/script_plugins/webidl_must_inherit.rs]: How about using pattern matching here 👀?

```rust
for def in idl {
  match def {
    weedle::Definition::Interface(def) | weedle::Definition::CallbackInterface(def) => {
      if let Some(parent) = def.inheritance {
        inherits = parent.identifier.0;
        break;
      }
    },
    _ => continue
  }
}
```

Or, maybe we can use `filter_map` with `take` here? 🤔

```rust
let inherits =
  idl.iter().filter_map(|def| {
    match def {
      weedle::Definition::Interface(def) | weedle::Definition::CallbackInterface(def) => def.inheritance,
      _ => None
    }
  })
  .take(1)
  .map_or("", |parent| parent.identifier.0);
```

Comment (Eijebong) [reply components/script_plugins/webidl_must_inherit.rs]: I don't think I can because the types of both `dep` is different

Comment (CYBAI) [reply components/script_plugins/webidl_must_inherit.rs]: Ah, I see; then please ignore me, thanks 🙇

**Comment by jdm** [comment]
@bors-servo retry
* #23375 23290

**Pull request merged**